### ISSUE TITLE
create new indicator based on number with at least one observation of a species within a deployment

### ISSUE THREAD

**Issue by jimcasaer** [inbo/camtraptor#184]
- somewhere between the RAI (#obs / #days of effort within the deployment) and raw occ (number of deployments having at least on obs of the species / total number of deployments) there is the #days with at least one observation (of a species) / #days of camera trap effort within the deployment => this ranges from 0 tot 1 (each day of the trapping effort) 
- given that this indicator can be calculated for each deployment this could easily be integrated as a feature in map_dep()

**Comment by damianooldoni** [comment]
Nice feature. Yes, it's possible to visualise it of course. 

We can use number of days, but we could use hours, weeks or months as well. So, days should be the default value of an argument.

Now, the most difficult stuff: naming. As this function measures the proportion of active days/hrs/weeks/months, I think to a name like `get_active_time()`. @jimcasaer, @peterdesmet ?

**Comment by peterdesmet** [comment]
_Suggested in camtraptor July 2023 coding sprint_

```R
active_time(
  package = NULL,
  group_by = "deploymentID" # new param, not sure if other groups make sense?
)
```

The returned information would be:

```ini
deploymentID
active_time
```

**Comment by damianooldoni** [comment]
Speaking about this with @PietrH. I agree with him that the name `active_time` I proposed long time ago is quite misleading. If somebody speaks about active time in camera trap context, we would immediately think about the time a camera was turned on (more or less the effort). Here, we speak about the number of days a camera detected a specific species.
More than active time I would speak about **detect time**.

This indicator can be a new column to the df returned by `summarize_observations()` (#367) as `detect_days`. 

I prefer to add this at a later stage (milestone v1.x) to avoid a possible bottleneck while releasing v1.0.

@peterdesmet, @jimcasaer, @martijn-bollen, @PietrH: what's your opinion about this new name?

**Comment by PietrH** [comment]
I prefer detect over active, it's more intuitive to me. But if others prefer active that's fine too!

**Comment by peterdesmet** [comment]
Yes, I'm in favour of `detect_days`, but then it should be in days and not be a percentage as originally asked by @jimcasaer. I actually think that is easier to understand.

So not:

deploymentID | scientificName | detect_percentage
--- | --- | ---
dep 1 | Sus scrofa | 0.75
dep 2 | Vulpes vulpes | 0.25

But rather:

deploymentID | scientificName | deployment_days | detect_days
--- | --- | --- | ---
dep 1 | Sus scrofa | 4 | 3
dep 1 | Vulpes vulpes | 4 | 1

`deployment_days` is derived from `deployment_duration` in `summarize_deployments()`

**Comment by damianooldoni** [comment]
Thanks @PietrH and @peterdesmet. I am happy you agree on `detect_*` approach. 
@peterdesmet: you wrote the output of this as a new function, while we are going in the direction of having `summarize_observations()`. I think we should not implement it as a new function, but as one(!) new column in output of `summarize_observations()` as we do with `rai`, which is a quite similar indicator. Then it makes sense to have `detect_percentage`?

**Comment by peterdesmet** [comment]
Yep, I meant it as part of `summarize_observations()`, but my preference would be as 2 columns, but that choice (`deployment_days` + `detect_days` or a single `detect_percentage` is up to @jimcasaer and @martijn-bollen)

**Comment by martijn-bollen** [comment]
I agree with @peterdesmet that `detect_days` and the `deployment_days` should be included -- these are the easiest to understand, at the same time the other proposed metrics can be derived from them, for: 

> deploymentID | scientificName | deployment_days | detect_days
> --- | --- | --- | ---
> dep 1 | Sus scrofa | 4 | 3
> dep 1 | Vulpes vulpes | 4 | 1
> dep 2 | Sus scrofa | 5 | 2
> dep 2 | Vulpes vulpes | 5 | 0

- naive occupancy as `sum(detect_days > 1) / n_distinct(deploymentID)` after running `summarize_observations(x, extend = TRUE)` such that deployments without observations (`detect_days = 0`) for a given species are included in the table.
- `detect_percentage` as `detect_days / deployment_days` either per deployment or across all deployments.

Would be nice to have this in v1.1.
